opm/simulators/wells/WellFilterCake.cpp

```diff
@@ -140,15 +140,29 @@ updateSkinFactorsAndMultipliers(const WellInterfaceGeneric<Scalar, IndexTraits>&
     const auto conc = ws.filtrate_conc;
     const std::size_t np = well_state.numPhases();
 
+    // Compute xflow-factor
+    Scalar xfact = 1.0;
+    Scalar qw_sum = 0.0;
+    Scalar qwpos_sum = 0.0;
+    for (int perf = 0; perf < nperf; ++perf) {
+        const auto crate = connection_rates[perf * np + water_index];
+        qwpos_sum += std::max(Scalar{0.}, crate);
+        qw_sum += crate;
+    }
+    if (qwpos_sum > qw_sum && qwpos_sum > 1.0e-12) {
```

Comment:
When things happens normally, the code is correct and fine, but a negative `xfact` is not desirable and it means erosion of the filter cake.    

We can either clamp the xfact to make sure it does not go negative with `xfact = std::max(Scalar(0.0), qw_sum) / qwpos_sum;`  or we `assert(xfact >= 0.)` .  I am struggling to decide which way we should go. `assert` can be too aggressive to terminate the running.  As a developer and user, what do you think?

Reply:
Good point, thanks. I think we should avoid assert, so I would prefer clamping. Will update!